### PR TITLE
Add login page

### PR DESCRIPTION
**Pull request by ilyayudovin**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -2,17 +2,24 @@
   "name": "trainee",
   "version": "0.1.0",
   "private": true,
+  "homepage": "http://ilyayudovin.github.io/spring/",
   "dependencies": {
+    "-": "0.0.1",
```
Comment (IAmVisco) [inline package.json]: Are you kidding me

File: package.json [hunk 2]
```diff
@@ -2,17 +2,24 @@
...
     "@testing-library/react": "^9.5.0",
     "@testing-library/user-event": "^7.2.1",
+    "prop-types": "latest",
     "react": "^16.13.1",
     "react-dom": "^16.13.1",
+    "react-hook-form": "^6.9.2",
+    "react-router-dom": "^5.2.0",
     "react-scripts": "3.4.3",
-    "prop-types": "latest"
+    "save-dev": "0.0.1-security"
```
Comment (IAmVisco) [inline package.json]: Same as above.

File: src/components/Login/Login.js [hunk 3]
```diff
@@ -0,0 +1,87 @@
+import React, { useState } from 'react';
+import {useHistory} from 'react-router-dom';
```
Comment (IAmVisco) [inline src/components/Login/Login.js]: ESlint broke.

File: src/components/Login/Login.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+  const onSubmit = (data) => {
+    const username = data.username;
+    const password = data.password;
+    if(username !== 'admin'){
+      setError("username", {
+        type: "wrong",
+        message: "username field is incorrect"
+      });
+    }
+    if(password !== '1234'){
```
Comment (IAmVisco) [inline src/components/Login/Login.js]: ESlint broke.

File: src/components/Login/Login.js [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    if (username === 'admin' && password === '1234') {
+      history.push('/home');
+    }
+  };
+
+  return (
+    <div className="loginContainer">
+      <div className="loginBox">
+        <h2>Login</h2>
+        <form onSubmit={handleSubmit(onSubmit)}>
```
Comment (IAmVisco) [inline src/components/Login/Login.js:40]: Weird naming chain.

File: src/components/Login/Login.js [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+          <div className="userBox">
+            <input
+              name='username'
+              ref={register({ required: true })}
+              type="text"
+              className={errors.username && "wrongInput" }
```
Comment (IAmVisco) [inline src/components/Login/Login.js]: Extra space.

File: src/components/Login/Login.js [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+            />
+            {errors.username?.type === "wrong" && <p className="errorMessage">username field is incorrect</p>}
```
Comment (IAmVisco) [inline src/components/Login/Login.js]: Capitalize first letter of the error message.

File: src/components/Login/Login.js [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+import hidePasswordIcon from './../../assets/icons8-hide-24.png';
+import './Login.scss';
+
+const Login = () => {
+  const { register, setError, errors, handleSubmit } = useForm();
+  const history = useHistory();
+
+  const [showPassword, setShowPassword] = useState(false);
+
+  const handlePassword = () => {
```
Comment (IAmVisco) [inline src/components/Login/Login.js]: `handle`?

**Review by IAmVisco — CHANGES_REQUESTED** [review]

**Comment by IAmVisco** [comment]
Has 0 security whatsoever.
Add logout button.

File: src/components/Login/Login.js [hunk 9]
```diff
@@ -8,50 +8,48 @@ import './Login.scss';
 const Login = () => {
   const { register, setError, errors, handleSubmit } = useForm();
   const history = useHistory();
-
   const [showPassword, setShowPassword] = useState(false);
 
-  const handlePassword = () => {
+  const passwordVisibility = () => {
```
Comment (IAmVisco) [inline src/components/Login/Login.js]: Missing a verb. Functions are actions, actions are usually represented as verbs.

File: src/components/Login/Login.js (hunk 5 above)
Comment (IAmVisco) [reply src/components/Login/Login.js:40]: Still not fixed.

File: src/components/Navigation/Navigation.js [hunk 10]
```diff
@@ -8,11 +9,16 @@ import MobileNavElement from '../MobileNavElement/MobileNavElement';
...
 const Navigation = () => {
   const [showMobileNav, setShowMobileNav] = useState(false);
-
+  const history = useHistory();
   const handleClick = () => {
     setShowMobileNav(!showMobileNav);
   };
 
+  const logOut = () => {
+    localStorage.setItem('isAuthorized', 'false');
```
Comment (IAmVisco) [inline src/components/Navigation/Navigation.js]: Usually done by just deleting entry.

File: src/index.js [hunk 11]
```diff
@@ -1,18 +1,32 @@
...
-      <Route exact path="/" component={Login} />
-      <Route exact path="/login" component={Login} />
-      <Route exact path="/home" component={Home} />
-    </Switch>
-  </BrowserRouter>
-);
+const App = () => {
+
+  const isLoggedIn = () => {
+    console.log('false');
```
Comment (IAmVisco) [inline src/index.js]: Do not commit console logs. Add esling rule to disallow console logs whatsoever. (I thought Airbnb ruleset has it?)

File: src/index.js [hunk 12]
```diff
@@ -1,18 +1,32 @@
...
+     <Switch>
+       <Route exact path="/" component={Login}/>
+       <Route exact path="/login" component={Login}/>
+       <Route render={() => (
+         !isLoggedIn() ? (
+           <Redirect to="/login"/>
+         ) : (
+           <Home />
+         )
+       )}/>
```
Comment (IAmVisco) [inline src/index.js]: I think it's not the correct flow. Create a component called `<AuthorizedRoute />` and do checks there. There are a ton of tutorials for such route wrappers.

**Review by IAmVisco — CHANGES_REQUESTED** [review]

File: src/components/Login/Login.js [hunk 13]
```diff
@@ -10,7 +10,7 @@ const Login = () => {
   const history = useHistory();
   const [showPassword, setShowPassword] = useState(false);
 
-  const passwordVisibility = () => {
+  const ChangePasswordVisibility = () => {
```
Comment (IAmVisco) [inline src/components/Login/Login.js]: Function names are written starting with lower case letters. Capitalized function are React components.

**Review by IAmVisco — CHANGES_REQUESTED** [review]

File: src/components/Login/Login.js (hunk 13 above)
Comment (ilyayudovin) [reply src/components/Login/Login.js]: Oops

**Pull request merged**